File: backtracking/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = [[]]

        for num in nums:
            res += [curr + [num] for curr in res]
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [1,2,3]
    print(s.subsets(nums))

File: Recursion/Subsets.cpp
class Solution {
    public:
        vector<vector<int>> subsets(vector<int>& nums) {
            vector<vector<int>> res;
            vector<int> path;
            backtrack(nums, 0, path, res);
            return res;
        }
        void backtrack(vector<int>& nums, int start, )
    };